### PR TITLE
Update tuner doc style

### PR DESCRIPTION
**Pull request by liuzhe-lz**
### Description ###

Mostly according to Tuesday meeting:

1. Fix style issues in tuner docs
2. Remove tuner built-in name in overview page
3. Split TPE & Random & Anneal
4. Rename titles from "Xxx Tuner in NNI" to "Xxx Tuner"
5. Replace `builtinTunerName` with `name` (for v2 config)

No sentence change in this PR.

Merge this first should make it easier to review #4339

### Checklist ###
  - [ ] ~~test case~~
  - [ ] ~~doc~~

### CODE REVIEW

**Review by ultmaster — APPROVED** [review]

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ Currently, we support the following algorithms:
 Usage of Built-in Tuners
```
Comment (QuanluZhang) [inline docs/en_US/Tuner/BuiltinTuner.rst:49]: so we still keep the content below the table?

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 2]
```diff
@@ -58,98 +58,56 @@ Note: Please follow the format when you write your ``config.yml`` file. Some bui
...
 
 Batch Tuner
 ^^^^^^^^^^^
 
-..
-
-   Built-in Tuner Name: BatchTuner
-
-
 **Suggested scenario**
```
Comment (QuanluZhang) [inline docs/en_US/Tuner/BuiltinTuner.rst:98]: do we need this line?

File: docs/en_US/Tuner/BuiltinTuner.rst [hunk 3]
```diff
@@ -221,11 +156,6 @@ As a strategy in a Sequential Model-based Global Optimization (SMBO) algorithm,
...
 ^^^^^^^^^
 
-..
-
-   Built-in Tuner Name: **PBTTuner**
-
-
 **Suggested scenario**
 
 Population Based Training (PBT) bridges and extends parallel search methods and sequential optimization methods. It requires relatively small computation resource, by inheriting weights from currently good-performing ones to explore better ones periodically. With PBTTuner, users finally get a trained model, rather than a configuration that could reproduce the trained model by training the model from scratch. This is because model weights are inherited periodically through the whole search process. PBT can also be seen as a training approach. If you don't need to get a specific configuration, but just expect a good model, PBTTuner is a good choice. `See details <./PBTTuner.rst>`__
```
Comment (QuanluZhang) [inline docs/en_US/Tuner/BuiltinTuner.rst:161]: we have two types of hyper link: "Detailed Description" and "See details"

File: docs/en_US/Tuner/HyperoptTuner.rst [hunk 4]
```diff
@@ -1,89 +1,15 @@
-TPE, Random Search, Anneal Tuners on NNI
-========================================
-
-TPE
```
Comment (QuanluZhang) [inline docs/en_US/Tuner/HyperoptTuner.rst:4]: the files for TPE and Random search are missing?

File: docs/en_US/Tuner/BuiltinTuner.rst (hunk 1 above)
Comment (cruiseliu) [reply docs/en_US/Tuner/BuiltinTuner.rst:49]: The sentences do not appear anywhere else. They discuss in which case a tuner should be chosen.
So it should be a separate work to rearrange these contents.

File: docs/en_US/Tuner/BuiltinTuner.rst (hunk 2 above)
Comment (cruiseliu) [reply docs/en_US/Tuner/BuiltinTuner.rst:98]: I don't know. It's not a style issue and has not been discussed.

File: docs/en_US/Tuner/BuiltinTuner.rst (hunk 3 above)
Comment (cruiseliu) [reply docs/en_US/Tuner/BuiltinTuner.rst:161]: Seems not a big problem. Don't want to touch them now.

File: docs/en_US/Tuner/HyperoptTuner.rst (hunk 4 above)
Comment (cruiseliu) [reply docs/en_US/Tuner/HyperoptTuner.rst:4]: Oh, seems I forgot to commit them.
Should not be a big problem because #4339 has included and updated them.

**Review by QuanluZhang — APPROVED** [review]

**Pull request merged**